### PR TITLE
fix(llamafarm-llama): honor temperature=0 as greedy, wire up repeat_penalty

### PR DESCRIPTION
**Pull request by mhamann**
## Summary

Two sampler-chain bugs in `llamafarm_llama.Llama._create_sampler`:

### 1. `temperature=0` was not greedy
When callers passed `temperature=0` the temperature sampler was skipped, but the `dist` (random) sampler was still appended to the chain — so generation became **stochastic sampling over whatever top_k/top_p/min_p left behind** instead of argmax.

On small models like Gemma 3 270M this produces near-garbage output for deterministic prompts. Example symptom I hit while chasing a bug on a drone deployment:
- Prompt: `"The capital of France is"` (temperature=0)
- Expected: `" Paris."`
- Actual: `" bal eccadiapolepolepolepole,pole..."`

**Fix:** when `temperature <= 0`, short-circuit with `llama_sampler_init_greedy()` and skip every other filter.

### 2. `repeat_penalty` was a no-op
The argument was accepted by every completion entry point (signature default `1.1`) but never translated into a `sampler_init` call, so the documented default never applied. Unfortunate because callers were under the impression they had a repetition penalty when they didn't.

**Fix:** wire it through `llama_sampler_init_penalties()` when `repeat_penalty != 1.0`.

## Test plan
- Added 4 regression tests in `tests/test_llama.py` using a mock llama lib to verify the sampler chain order for each sampling mode.
- All 21 existing tests still pass.

```
tests/test_llama.py::TestSamplerChain::test_temperature_zero_uses_greedy_only PASSED
tests/test_llama.py::TestSamplerChain::test_positive_temperature_uses_stochastic_chain PASSED
tests/test_llama.py::TestSamplerChain::test_repeat_penalty_wired_into_chain PASSED
tests/test_llama.py::TestSamplerChain::test_repeat_penalty_one_is_noop PASSED
```

## Related
This was chased down while debugging a Gemma 3 fine-tune that worked perfectly via `transformers` fp32 and standard `llama-cpp-python` but produced garbage through `llamafarm_llama` on the same GGUF. The sampler chain was the only material behavioral difference.

### CODE REVIEW

**Comment by mhamann** [comment]
Addressed the cubic / qodo feedback in ca392216.

Both bots flagged the same real bug: the `temperature <= 0` greedy early-return ran before the `repeat_penalty` wiring, so the public default `repeat_penalty=1.1` was silently a no-op in greedy mode.

**Fix:** moved the penalties sampler above the greedy early-return, so the chain is now `penalties -> greedy` when both are active and `penalties -> top_k/top_p/min_p -> temp -> dist` otherwise. `llama_sampler_init_penalties` is a valid standalone chain element in llama.cpp, so prepending it to a greedy chain still produces deterministic argmax output — just over penalized logits.

Added `test_temperature_zero_with_repeat_penalty_applies_penalty_then_greedy` asserting the `["penalties", "greedy"]` chain order, and tightened `test_temperature_zero_uses_greedy_only` to pass `repeat_penalty=1.0` so its narrow intent is preserved. All 22 tests pass.
